test/dotnet-project-add.Tests/DotnetProjectAddTests.cs

```diff
@@ -0,0 +1,174 @@
+            .Execute()
+            .Should().Fail()
+            .And.HaveStdErrContaining("The target directory already exists");
+    }
+
+    [Fact]
+    public void MultipleEntryPointFiles()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program1.cs"), "Console.WriteLine(1);");
+        File.WriteAllText(Path.Join(testInstance.Path, "Program2.cs"), "Console.WriteLine(2);");
+
+        new DotnetCommand(Log, "project", "add", "Program1.cs")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Pass();
+
+        new DirectoryInfo(testInstance.Path)
+            .EnumerateDirectories().Select(d => d.Name).Order()
+            .Should().BeEquivalentTo(["Program1"]);
+
+        new DirectoryInfo(Path.Join(testInstance.Path, "Program1"))
+            .EnumerateFileSystemInfos().Select(f => f.Name).Order()
+            .Should().BeEquivalentTo(["Program1.csproj", "Program1.cs"]);
+    }
+
+    [Fact]
+    public void NoFileArgument()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+
+        new DotnetCommand(Log, "project", "add")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Fail()
+            .And.HaveStdErrContaining("Required argument missing for command");
+    }
+
+    [Fact]
+    public void NonExistentFile()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+
+        new DotnetCommand(Log, "project", "add", "NotHere.cs")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Fail()
+            .And.HaveStdErrContaining("The specified file must exist");
+    }
+
+    [Fact]
+    public void NonCSharpFile()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.vb"), "");
+
+        new DotnetCommand(Log, "project", "add", "Program.vb")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Fail()
+            .And.HaveStdErrContaining("The specified file must exist and have '.cs' file extension");
+    }
+
+    [Fact]
+    public void ExtensionCasing()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.CS"), "Console.WriteLine();");
+
+        new DotnetCommand(Log, "project", "add", "Program.CS")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Pass();
+
+        new DirectoryInfo(Path.Join(testInstance.Path, "Program"))
+            .EnumerateFileSystemInfos().Select(f => f.Name).Order()
+            .Should().BeEquivalentTo(["Program.csproj", "Program.CS"]);
+    }
+
+    [Theory]
+    [InlineData("")]
+    [InlineData("class C;")]
```

Comment:
Consider testing also with a variety of `.cs` file names.

Reply:
There are actually several different file names used throughout the tests like `Program.CS`, `app/Program.cs`, `MyApp.cs`. Not sure if something else needs to be tested.